### PR TITLE
Included AWS Elastic Beanstalk deployment configuration scripts

### PR DESCRIPTION
**Pull request by sandeepmjay**
Included AWS Elastic Beanstalk configurations. Changes include
- AMI creation for node.js environment
- Config files for pre/during/post deployment execution
- Customizations to apache proxy configurations.

### CODE REVIEW

File: .ebextensions/01_install_httpd.config [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#
+# STEP 1:
+# Create the httpd proxy config file to replace the default configuration
+#
+files:
+
```
Comment (johncrossman) [inline .ebextensions/01_install_httpd.config]: @sandeepmjay I'd expect no extra line breaks in this config structure. Do you have an opinion on a good standard?

File: .ebextensions/00_ami.config [hunk 2]
```diff
@@ -0,0 +1,5 @@
+option_settings:
+  aws:elasticbeanstalk:container:nodejs:
+    NodeVersion: 6.2.2
+    ProxyServer: apache
+    NodeCommand: "npm start"
```
Comment (johncrossman) [inline .ebextensions/00_ami.config]: @sandeepmjay speaking of standards, I prefer the "lightness" of single quotes.

File: .ebextensions/01_install_httpd.config [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      # using Configuration Files.
+      # http://docs.amazonwebservices.com/elasticbeanstalk/latest/dg/customize-containers.html
+      #
+      # Modifications of httpd.conf can be performed using container_commands to modify the staged version
+      # located in /tmp/deployment/config/etc#httpd#conf#httpd.conf
+      # This file was overwritten during deployment
+      # by .ebextensions/02_container_commands.config
+
+      Listen 8080
+      AddOutputFilterByType DEFLATE text/plain text/css application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript
```
Comment (johncrossman) [inline .ebextensions/01_install_httpd.config]: @sandeepmjay  we don't need `text/html` in line above?

File: .ebextensions/02_container_commands.config [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#
+# STEP 2:
+# Container commands can be used to run commands during deployment.
+# Run the script to overwrite the httpd config template in /tmp/deployment/config.
+#
+container_commands:
+
+  01_runmyshellcommand:
```
Comment (johncrossman) [inline .ebextensions/02_container_commands.config]: @sandeepmjay can we use a more descriptive name?  E.g., `01_apache_config`

File: .ebextensions/99_delayedscript.config [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#
+# STEP 3:
+# Post deployment elastic beanstalk deploys the commands
+# in /opt/elasticbeanstalk/hooks/appdeploy/post. The post folder
+# needs to be created as it is not part of default awd beanstalk deployment.
+# The config creates a post deployment scripts that is required to run the app
+
+commands:
+  create_post_dir:
+    command: "mkdir /opt/elasticbeanstalk/hooks/appdeploy/post"
```
Comment (johncrossman) [inline .ebextensions/99_delayedscript.config]: @sandeepmjay Would `mkdir -p ...` be quieter and/or necessary? I know that `mkdir` without `-p` will complain - not abort - if the dir already exists.

**Comment by johncrossman** [comment]
@sandeepmjay I'm done with review. Thanks for walking me through this earlier. I'll merge once I hear back on some rather minor concerns.

File: .ebextensions/99_delayedscript.config [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#
+# STEP 3:
+# Post deployment elastic beanstalk deploys the commands
+# in /opt/elasticbeanstalk/hooks/appdeploy/post. The post folder
+# needs to be created as it is not part of default awd beanstalk deployment.
```
Comment (paulkerschen) [inline .ebextensions/99_delayedscript.config]: @sandeepmjay typo `awd`

File: .ebextensions/01_install_httpd.config (hunk 1 above)
Comment (sandeepmjay) [inline .ebextensions/01_install_httpd.config]: Fixed

File: .ebextensions/00_ami.config (hunk 2 above)
Comment (sandeepmjay) [inline .ebextensions/00_ami.config]: Switched to single quotes. Amazon examples seems to use double quotes everywhere. However I have tested it with single quote and found no apparent green gremlins.

File: .ebextensions/01_install_httpd.config (hunk 3 above)
Comment (sandeepmjay) [inline .ebextensions/01_install_httpd.config]: added

File: .ebextensions/02_container_commands.config (hunk 4 above)
Comment (sandeepmjay) [inline .ebextensions/02_container_commands.config]: Fixed

File: .ebextensions/99_delayedscript.config (hunk 5 above)
Comment (sandeepmjay) [inline .ebextensions/99_delayedscript.config]: Added -p

File: .ebextensions/99_delayedscript.config (hunk 6 above)
Comment (sandeepmjay) [inline .ebextensions/99_delayedscript.config]: fixed

**Comment by sandeepmjay** [comment]
@johncrossman @paulkerschen Thanks for the review !

**Pull request merged**